### ISSUE TITLE
Elements should be validated before starting to write

### ISSUE THREAD

**Issue by timtreis** [scverse/spatialdata#804]
**Is your feature request related to a problem? Please describe.**
I am writing a large sdata object I created to disk. It wrote the image for 12 min and then failed because my `adata.obs[region_key]` was not categorical. 

**Describe the solution you'd like**
All the validations should happen before we start writing.

**Comment by LucaMarconato** [comment]
I totally agree, thanks for reporting. It may be fixed by https://github.com/scverse/spatialdata/pull/703, I will check this when merging that PR.
